### ISSUE TITLE
Remove test.py and calc_all_test()? 

### ISSUE THREAD

**Issue by MattHJensen** [open-source-economics/Tax-Calculator#470]
from #469:

> there are some differences between calc_all() and calc_all_test() in calculate.py.

> test.py calls calc_all_test(), but every other application of the calculator that I know of uses call_all(). For one thing, calc_all_test() does not execute the proper itemized filing optimization. It also doesn't allow for the itemized deduction surtax.

My suggestion is that we get rid of `test.py` and `calc_all_test()` so we don't need to worry about maintaining them. 

What do others think about this?

**Comment by martinholmer** [comment]
@MattHJensen said:
> My suggestion is that we get rid of test.py and calc_all_test() so we don't need to worry about maintaining them.

If you do that, what is somebody who wants to run puf.csv through the Tax-Calculator supposed to do?

**Comment by MattHJensen** [comment]
> If you do that, what is somebody who wants to run puf.csv through the Tax-Calculator supposed to do?

We could add a script that uses a calculator in a more typical style and that takes advantage of the calc_to_ascii.py that @xiyuw123 has waiting for review in the open PR #353. 

This would look something like this: 

```
from taxcalc import Policy, Records, Calculator
puf = Records()
clp = Policy()
calc = Calculator(policy=clp, records=puf)
calc.calc_all()
calc_to_ascii(calc, ascii_results="ascii_results.txt")
```

As an enhancement, we could also allow users to specify non-default input files or output filename from the command line. 

What do you think?

**Comment by martinholmer** [comment]
@MattHJensen said:
> We could add a script that uses a calculator in a more typical style and that takes advantage of the calc_to_ascii.py that @xiyuw123 has waiting for review in the open PR #353.
>
> This would look something like this:
```
from taxcalc import Policy, Records, Calculator
puf = Records()
clp = Policy()
calc = Calculator(policy=clp, records=puf)
calc.calc_all()
calc_to_ascii(calc, ascii_results="ascii_results.txt")
```
>
> As an enhancement, we could also allow users to specify non-default input files or output filename from the command line.
>
> What do you think?

So, the ```test.py``` script would be revised to look like the code above?  Right?  If so, then this seems like a reasonable approach that cuts down on the amount of code to maintain.  Personally, I don't think it is worth it to do the command-line stuff; anybody who wants to read in a file named something other than ```puf.csv``` or get the ascii output in a file named something other than ```ascii_results.txt``` can just temporarily edit the ```test.py``` file (or rename files a the command-line prompt).

After glancing at the calc_to_ascii.py file, I wonder about at least three things:
(1) why isn't this function in the utils.py file? (Don't think it warrants a separate file.)
(2) why do we have a fixed 15 characters for each field?  (This is going to generate a HUGE file.)
(3) why the tab-separated format?  (Simply use space-separated columns like the other output files.)

**Comment by MattHJensen** [comment]
> So, the test.py script would be revised to look like the code above? Right?

That is right. 

> Personally, I don't think it is worth it to do the command-line stuff; anybody who wants to read in a file named something other than puf.csv or get the ascii output in a file named something other than ascii_results.txt can just temporarily edit the test.py file (or rename files a the command-line prompt).

Makes sense to me too. 

> After glancing at the calc_to_ascii.py file, I wonder about at least three things:

@GoFroggyRun is making a version of test.py that writes to csv for his testing against taxcalc.sas rather than using calc_to_ascii.py. @GoFroggyRun, could you submit that file as a PR once it is ready?

**Comment by martinholmer** [comment]
@MattHJensen said:
> @GoFroggyRun is making a version of test.py that writes to csv for his testing against taxcalc.sas rather than using calc_to_ascii.py. @GoFroggyRun, could you submit that file as a PR once it is ready?

Sounds like a good plan.

**Comment by GoFroggyRun** [comment]
@MattHJensen Sure. The script should be working, and I'll open a PR right away. But one thing is that the csv file might be huge, which might only allow the validation against sas calculator at this moment.

**Comment by MattHJensen** [comment]
Resolved with #472. Closing.
